tests/integration/README.md

```diff
@@ -35,3 +35,14 @@ For places where more targets are returned than expected, the subclass should ov
 For places where a target determinator has a different, equally valid, interpretation of what should be returned, the test method can simply be overridden.
 
 For places where behavior is not supported, or simply incorrect, the overridden test method should be annotated with an `@Ignore` annotation, with an explanation of why.
+
+## Adding new tests
+
+Adding tests is slightly fiddly, as it involves making coordinated changes across the testdata repo and this one.
+
+This is roughly the flow to add tests:
+1. Decide whether this is specific to this TD implementation, or applies generally to all target determinators. If it's general, the new test should go in [java/com/github/bazel_contrib/target_determinator/integration/Tests.java], otherwise in [java/com/github/bazel_contrib/target_determinator/integration/TargetDeterminatorSpecificFlagsTest.java] - in both cases, the existing tests should be easy to crib from.
+1. If new commits are needed in the testdata repo (which is most often the case), clone that, and add commits to it. Each test commit is typically in a unique branch based on https://github.com/bazel-contrib/target-determinator-testdata/commit/6682820b4acb455f13bc3cf8f7d254056092e306 - we try to have each branch have the minimal changes needed for each test, rather than amassing lots of unrelated changes on fewer branches.
```

Comment:
Maybe mention that an immutable tag needs to be created too?

Reply:
Done!